src/main/java/software/amazon/cloudformation/proxy/package-info.java

```diff
@@ -0,0 +1,461 @@
+/**
+ * This package provide facilities to make it easy to work against AWS APIs that
+ * are eventually consistent for applying resource state. Developers need to
+ * sequence APIs calls in order to effectively apply state. When dependent
+ * resources are not available, e.g. when associating KMS Key with CloudWatch
+ * Log group, the key might not have propogated to region leading to failures to
+ * associate with the log group. This framework provides developers the
+ * facilities to deal with all of these situations using a simple functional
+ * model.
+ * <p>
+ * The framework handles the following for developers:
+ * <ol>
+ * <li>Taking care of communicating all progress events consistently with
+ * CloudFormation including appropriate handshakes</li>
+ * <li>Handles all client side error reporting correctly with appropriate
+ * failure codes and errors messages</li>
+ * <li>Handles common AWS service exceptions like Throttle and several other
+ * retry errors based on HTTP codes</li>
+ * <li>Can auto-renew STS token expiry transparently if needed or schedule
```

Comment:
Does it actually do this? How? And should it be?

Reply:
I have an implementation that does a refresh. This is used when we run it locally. Today AccessDenied with Expired Token can be converted to progress event with callback > 0s. Separated that into a different CR.